src/middleware.ts

```diff
@@ -248,7 +248,10 @@ export const persist = <S extends State>(
       blacklist.forEach((key) => delete state[key])
     }
 
-    return storage?.setItem(name, await serialize({ state, version }))
+    let serializeResult = serialize({ state, version })
+    return serializeResult instanceof Promise
+      ? serializeResult.then((value) => storage?.setItem(name, value))
+      : storage?.setItem(name, serializeResult)
```

Comment:
To keep the behavior, we don't invoke serialize, if storage is undefined.
```suggestion
    if (storage) {
      const serializeResult = serialize({ state, version })
      return serializeResult instanceof Promise
        ? serializeResult.then((value) => storage.setItem(name, value))
        : storage.setItem(name, serializeResult)
    }
```

Reply:
pretty neat.